cedar-policy-core/src/ast/entity.rs

```diff
@@ -28,38 +28,61 @@ use serde::{Deserialize, Serialize};
 use serde_with::{serde_as, TryFromInto};
 use smol_str::SmolStr;
 use std::collections::{BTreeMap, HashMap, HashSet};
+use std::str::FromStr;
 use thiserror::Error;
 
-/// We support two types of entities. The first is a nominal type (e.g., User, Action)
-/// and the second is an unspecified type, which is used (internally) to represent cases
-/// where the input request does not provide a principal, action, and/or resource.
+/// Entity type names are just [`Name`]s, but we have some operations on them specific to entity types.
 #[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone, Hash, PartialOrd, Ord)]
 #[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
-pub enum EntityType {
-    /// Concrete nominal type
-    Specified(Name),
-    /// Unspecified
-    Unspecified,
-}
+#[serde(transparent)]
+pub struct EntityType(Name);
 
 impl EntityType {
     /// Is this an Action entity type
     pub fn is_action(&self) -> bool {
-        match self {
-            Self::Specified(name) => name.basename() == &Id::new_unchecked("Action"),
-            Self::Unspecified => false,
-        }
+        self.0.basename() == &Id::new_unchecked("Action")
+    }
+
+    /// The name of this entity type
+    pub fn name(&self) -> &Name {
+        &self.0
+    }
+
+    /// Get the source location for this Entity Type
```

Comment:
For consistency with the docstring on `pub fn name`
```suggestion
    /// The source location of this entity type
```